src/collection.js

```diff
@@ -186,3 +191,85 @@ function compareValues(a, b, descending) {
     return descending ? -1 : 1;
   }
 }
+
+function makeMapFn(projection) {
```

Comment:
I'd prefer if this function just took in a doc and applied a projection, rather than returning a function that applies the projection.

I know sift follows the "return a function that applies the query" pattern. But I disagree with their decision to use that pattern, it makes the code harder to follow and doesn't provide much benefit.